### PR TITLE
Feature/jl/hardening

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/platform/Server.java [hunk 1]
```diff
@@ -117,6 +117,8 @@
   /**
    * @return the server URL
    * @since since 2.4 on batch side only, since 5.6 on both batch side and server side (WebServer and Compute Engine)
+   * @deprecated since 6.5, please use {@link #getPublicRootUrl()} instead
    */
+  @Deprecated
   public abstract String getURL();
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/platform/Server.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java [hunk 2]
```diff
@@ -86,15 +87,23 @@ public ComponentDto getByKey(DbSession dbSession, String key) {
   }
 
   private ComponentDto getByKey(DbSession dbSession, String key, String label) {
-    return checkFoundWithOptional(dbClient.componentDao().selectByKey(dbSession, key), "%s key '%s' not found", label, key);
+    return checkComponent(() -> dbClient.componentDao().selectByKey(dbSession, key), () -> format("%s key '%s' not found", label, key));
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java]: Will it work ?

File: server/sonar-server/src/test/java/org/sonar/server/component/ws/ShowActionTest.java [hunk 3]
```diff
@@ -138,7 +136,7 @@ public void show_with_browse_permission() {
   @Test
   public void show_provided_project() {
     userSession.logIn().setRoot();
-    componentDb.insertComponent(newPrivateProjectDto(db.organizations().insert(), "project-uuid").setEnabled(false));
+    componentDb.insertComponent(newPrivateProjectDto(db.organizations().insert(), "project-uuid"));
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/component/ws/ShowActionTest.java]: no

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java [hunk 4]
```diff
@@ -132,7 +141,7 @@ private ComponentDto checkIsProject(ComponentDto component) {
   public OrganizationDto getOrganization(DbSession dbSession, ComponentDto component) {
     String organizationUuid = component.getOrganizationUuid();
     return dbClient.organizationDao().selectByUuid(dbSession, organizationUuid)
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java]: let's discuss

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java]: Forget my "let's discusss", I found you were talking of new code. 
Totally agree to use  checkFoundWithOptional here

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java (hunk 2 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentFinder.java]: forget, it's ok

**Pull request merged**